services/workflows-service/src/auth/local/local-auth.guard.ts

```diff
@@ -1,14 +1,35 @@
 import { AuthGuard } from '@nestjs/passport';
 import { ExecutionContext } from '@nestjs/common';
 import type { Request } from 'express';
+import { createClient } from '@supabase/supabase-js';
+import { env } from '@/env';
 
 export class LocalAuthGuard extends AuthGuard('local') {
   async canActivate(context: ExecutionContext) {
     const result = await super.canActivate(context);
     const request = context.switchToHttp().getRequest<Request>();
-
     await super.logIn(request);
-
+    if (env.TELEMETRY_ENABLED && env.TELEMETRY_SUPABASE_URL && env.TELEMETRY_SUPABASE_API_KEY) {
+      try {
+        const SupabaseClient = createClient(
+          env.TELEMETRY_SUPABASE_URL,
+          env.TELEMETRY_SUPABASE_API_KEY,
+          {
+            db: { schema: 'public' },
+          },
+        );
+        const fullUrl = `${request.protocol}://${request.get('Host')}${request.originalUrl}`;
+        const { data: result, error } = await SupabaseClient.from('logins').insert([
+          { url: fullUrl },
+        ]);
+        if (error) {
+          console.error('Error inserting data:', error.message);
+          return false;
+        }
+      } catch (err) {
+        console.error('Unexpected error:', err);
+      }
+    }
```

Comment:
Please create a proper Supabase client service and inject this service in here:

https://stackoverflow.com/questions/52862644/inject-service-into-guard-in-nest-js

Reply:
## Use Dependency Injection:

Inject SupabaseService into LocalAuthGuard via the constructor. This will make it easier to mock SupabaseService during testing.
Use ConfigService (provided by @nestjs/config) to manage environment variables, making it easier to handle configurations across the application.
Separate Real and Mock Supabase Implementations: